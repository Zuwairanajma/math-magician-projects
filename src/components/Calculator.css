.calculator {
  display: flex;
  justify-content: space-between;
  margin-top: 150px;
}

.calculator h2 {
  font-size: 2em;
  color: #858694;
}

.tentative-cal {
  width: 50%;
  min-width: 300px;
  box-shadow: 0 0 10px 0;
}

.display {
  text-align: right;
  background-color: #858694;

  /* padding: 1rem; */
  color: white;
  flex-direction: column;
  font-size: 2em;
  padding: 3% 1%;
}

.buttons-row {
  display: flex;
  justify-content: space-between;
}

.buttons-row button {
  width: 200px;
  height: 40px;
  background-color: #e0e0e0;
  font-size: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.buttons-row button:last-child {
  background-color: #f5913e;
}

.last-buttons {
  display: flex;
}

.last-buttons button:first-child {
  width: 424px;
}

@media (width <= 768px) {
  .calculator {
    flex-direction: column;
    align-items: center;
    gap: 2em;
  }
}
