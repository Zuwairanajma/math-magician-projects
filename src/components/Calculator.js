import { useState, React } from 'react';
import CalculatorButton from './Button';
import './Calculator.css';
import calculate from './logic/calculate';

function Calculator() {
  const [numbers, setNumbers] = useState({
    total: null,
    next: null,
    operation: null,
  });

  function handleClick(number) {
    const newNumber = calculate(numbers, number);
    setNumbers(newNumber);
  }

  return (
    <div id="calculator" className="calculator">

      <h2>Let&apos;s do some math!</h2>
      <div className="tentative-cal">
        <div className="display" data-testid="display">
          { numbers.total }
          { numbers.operation }
          { numbers.next }
        </div>
        <div className="buttons-row">
          <CalculatorButton label="AC" onClick={() => handleClick('AC')} />
          <CalculatorButton label="+/-" onClick={() => handleClick('+/-')} />
          <CalculatorButton label="%" onClick={() => handleClick('%')} />
          <CalculatorButton label="รท" onClick={() => handleClick('รท')} />
        </div>
        <div className="buttons-row">
          <CalculatorButton label="7" onClick={() => handleClick('7')} />
          <CalculatorButton label="8" onClick={() => handleClick('8')} />
          <CalculatorButton label="9" onClick={() => handleClick('9')} />
          <CalculatorButton label="x" onClick={() => handleClick('x')} />
        </div>
        <div className="buttons-row">
          <CalculatorButton label="4" onClick={() => handleClick('4')} />
          <CalculatorButton label="5" onClick={() => handleClick('5')} />
          <CalculatorButton label="6" onClick={() => handleClick('6')} />
          <CalculatorButton label="-" onClick={() => handleClick('-')} />
        </div>
        <div className="buttons-row">
          <CalculatorButton label="1" onClick={() => handleClick('1')} />
          <CalculatorButton label="2" onClick={() => handleClick('2')} />
          <CalculatorButton label="3" onClick={() => handleClick('3')} />
          <CalculatorButton label="+" onClick={() => handleClick('+')} />
        </div>
        <div className="buttons-row last-buttons">
          <CalculatorButton label="0" onClick={() => handleClick('0')} />
          <CalculatorButton label="." onClick={() => handleClick('.')} />
          <CalculatorButton label="=" onClick={() => handleClick('=')} />
        </div>
      </div>
    </div>
  );
}

export default Calculator;
