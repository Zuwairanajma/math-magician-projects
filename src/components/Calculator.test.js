import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/jest-globals';
import Calculator from './Calculator';

describe('Calculator component', () => {
  test('renders the component without errors', () => {
    const renderer = render(<Calculator />);
    expect(renderer).toMatchSnapshot();
  });

  test('displays the initial values correctly', () => {
    const { getByTestId } = render(<Calculator />);
    const displayElement = getByTestId('display');
    expect(displayElement.textContent).toBe('');
  });

  test('handles button clicks correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    const displayElement = getByTestId('display');

    fireEvent.click(getByText('1'));
    expect(displayElement.textContent).toBe('1');

    fireEvent.click(getByText('+'));
    expect(displayElement.textContent).toBe('1+');

    fireEvent.click(getByText('2'));
    expect(displayElement.textContent).toBe('1+2');

    fireEvent.click(getByText('='));
    expect(displayElement.textContent).toBe('3');

    fireEvent.click(getByText('AC'));
    expect(displayElement.textContent).toBe('');
  });
});
