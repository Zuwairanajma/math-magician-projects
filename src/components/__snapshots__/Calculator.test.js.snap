// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders the component without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator"
      >
        <div
          class="display"
          data-testid="display"
        />
        <div
          class="buttons-row"
        >
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="buttons-row"
        >
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="buttons-row"
        >
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="buttons-row"
        >
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="buttons-row last-buttons"
        >
          <button
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator"
    >
      <div
        class="display"
        data-testid="display"
      />
      <div
        class="buttons-row"
      >
        <button
          type="button"
        >
          AC
        </button>
        <button
          type="button"
        >
          +/-
        </button>
        <button
          type="button"
        >
          %
        </button>
        <button
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="buttons-row"
      >
        <button
          type="button"
        >
          7
        </button>
        <button
          type="button"
        >
          8
        </button>
        <button
          type="button"
        >
          9
        </button>
        <button
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="buttons-row"
      >
        <button
          type="button"
        >
          4
        </button>
        <button
          type="button"
        >
          5
        </button>
        <button
          type="button"
        >
          6
        </button>
        <button
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="buttons-row"
      >
        <button
          type="button"
        >
          1
        </button>
        <button
          type="button"
        >
          2
        </button>
        <button
          type="button"
        >
          3
        </button>
        <button
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="buttons-row last-buttons"
      >
        <button
          type="button"
        >
          0
        </button>
        <button
          type="button"
        >
          .
        </button>
        <button
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
